lib/spack/spack/environment/environment.py

```diff
@@ -2929,93 +2914,25 @@ def included_config_scopes(self) -> List[spack.config.ConfigScope]:
-                        if basename.endswith(".yaml"):
-                            config_path = os.path.join(config_path, basename)
-                    else:
-                        staged_path = spack.config.fetch_remote_configs(
-                            config_path, str(self.config_stage_dir), skip_existing=True
-                        )
-                        if not staged_path:
-                            raise SpackEnvironmentError(
-                                "Unable to fetch remote configuration {0}".format(config_path)
-                            )
-                        config_path = staged_path
-
-                elif include_url.scheme:
-                    raise ValueError(
-                        f"Unsupported URL scheme ({include_url.scheme}) for "
-                        f"environment include: {config_path}"
-                    )
-
-            # treat relative paths as relative to the environment
-            if not os.path.isabs(config_path):
-                config_path = os.path.join(self.manifest_dir, config_path)
-                config_path = os.path.normpath(os.path.realpath(config_path))
-
-            if os.path.isdir(config_path):
-                # directories are treated as regular ConfigScopes
-                config_name = f"env:{self.name}:{os.path.basename(config_path)}"
-                tty.debug(f"Creating DirectoryConfigScope {config_name} for '{config_path}'")
-                scopes.append(spack.config.DirectoryConfigScope(config_name, config_path))
-            elif os.path.exists(config_path):
-                # files are assumed to be SingleFileScopes
-                config_name = f"env:{self.name}:{config_path}"
-                tty.debug(f"Creating SingleFileScope {config_name} for '{config_path}'")
-                scopes.append(
-                    spack.config.SingleFileScope(
-                        config_name, config_path, spack.schema.merged.schema
-                    )
-                )
-            else:
-                missing.append(config_path)
-                continue
+        if not includes:
+            return []
 
-        if missing:
-            msg = "Detected {0} missing include path(s):".format(len(missing))
-            msg += "\n   {0}".format("\n   ".join(missing))
-            raise spack.config.ConfigFileError(msg)
+        # Load config scopes added via environment 'include:' in reverse so
+        # that highest-precedence scopes are last.
+        name_prefix = f"env:{self.name}"
+        scopes: List[spack.config.ConfigScope] = []
+        for entry in reversed(includes):
+            included_scope = spack.config.include_path_scope(
+                spack.config.included_path(entry),
+                name_prefix,
+                self.config_stage_dir,
+                self.manifest_dir,
+            )
+            if included_scope is not None:
+                scopes.append(included_scope)
```

Comment:
It shouldn't be necessary to do this here if `include:` is handled by the config system.  Just pushing the env's scope should handle any includes.  Similarly, if we ever overrode a scope with `config.override`, it shouldn't have to think about the scope's includes -- any push should just handle includes automatically.